### PR TITLE
LSP: Speed up publish diagnostics on project scan.

### PR DESCRIPTION
**Pull request by dbalek**
An attempt to speed up publishing diagnostics collected on project scan by directly publishing data from caches.

### CODE REVIEW

File: ide/parsing.indexing/apichanges.xml [hunk 1]
```diff
@@ -87,6 +87,20 @@ is the proper place.
 <!-- ACTUAL CHANGES BEGIN HERE: -->
 
   <changes>
+      <change id="ErrorsCache.getErrors">
+          <api name="IndexingAPI"/>
+          <summary>Added method to ErrorsCache to return all errors or warnings for the given file</summary>
+          <version major="9" minor="37"/>
```
Comment (sdedic) [inline ide/parsing.indexing/apichanges.xml:93]: I can't find spec version bump in `project.properties`

Comment (dbalek) [reply ide/parsing.indexing/apichanges.xml:93]: Oops, thanks for pointing this out.

**Review by lahodaj — APPROVED** [review]
Looks reasonable to me.

**Review by MartinBalin — APPROVED** [review]

**Comment by mbien** [comment]
this seems to be causing a regression https://github.com/apache/netbeans/issues/7981

**Pull request merged**